### PR TITLE
Extract callbacks to module

### PR DESCRIPTION
**Pull request by julianrubisch**
# Refactoring

## Description

Very simplistic refactoring to extract `callbacks` to a module, giving the `Reflex` class a bit more space and create a dedicated space for callback behavior.

Please advise on style, I simply wrapped the singleton class definition in a `class_eval` block, let me know if there's a more idiomatic way!

## Checklist

- [X] My code follows the style guidelines of this project
- [X] Checks (StandardRB & Prettier-Standard) are passing
- [X] This is not a documentation update

Please note that the best way to suggest changes or updates to the [documentation](https://docs.stimulusreflex.com) is to [join Discord](https://discord.gg/XveN625) and leave a note in the #docs channel. Any documentation updates posted as PRs cannot be accepted at this time. :heart:

### CODE REVIEW

File: lib/stimulus_reflex/callbacks.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+module StimulusReflex
+  module Callbacks
+    def self.included(klass)
```
Comment (hopsoft) [inline lib/stimulus_reflex/callbacks.rb:3]: Considering that we have Rails as a dependency, we may want to look at using `ActiveSupport::Concern` and benefit from the prettier semantics. Thoughts? https://api.rubyonrails.org/classes/ActiveSupport/Concern.html

Comment (julianrubisch) [reply lib/stimulus_reflex/callbacks.rb:3]: Will happily rewrite if this is the direction 👍🏻

Comment (julianrubisch) [reply lib/stimulus_reflex/callbacks.rb]: made the change 👍

**Review by andrewmcodes — APPROVED** [review]
Looks like a nice abstraction with the benefit of the AS enhancement. This code is more readable barring objections I say we get it in. 

However now that we have created this abstraction, would it be easier to add some test coverage? That would seem like a win to me. 

@hopsoft @leastbad

**Comment by leastbad** [comment]
I agree that this would be an improvement. The only reason I haven't merged it is because I don't want to short-circuit any initiatives that might be underway to add some test coverage, as Andrew suggested.

**Pull request merged**